src/main/java/edu/hawaii/its/api/service/UpdateTimestampService.java

```diff
@@ -21,18 +27,40 @@ public class UpdateTimestampService {
 
     public static final Log logger = LogFactory.getLog(MembershipService.class);
 
+    @Autowired
+    public GroupPathService groupPathService;
+
     public GroupingTimestampResult update(GroupingResult groupingResult) {
         if (groupingResult.getResultCode().equals("SUCCESS")) {
             return updateLastModifiedTimestamp(groupingResult.getGroupPath());
         }
         return new GroupingTimestampResult();
     }
 
+    public List<GroupingTimestampResult> updateBasePath(GroupingResult groupingResult) {
+        if (groupingResult.getResultCode().equals("SUCCESS")) {
+            List<String> groupPaths = new ArrayList<>();
+            groupPaths.add(groupingResult.getGroupPath());
+            groupPaths.add(groupPathService.getGroupingPath(groupingResult.getGroupPath()));
+            return updateLastModifiedTimestamp(groupPaths);
+        }
+        return Collections.emptyList();
+    }
```

Comment:
I just thought of a better idea. What if we have a single update function but the function is able to parse a groupingPath into a list or a singleton list. For example, if the groupingPath of the groupingResult is tmp:testiwta:testiwta-aux:owners, a list with [tmp:testiwta:testiwta-aux, tmp:testiwta:testiwta-aux:owners] is generated. If the groupingPath of the groupingResult is just tmp:testiwta:testiwta-aux, the list just will be [tmp:testiwta:testiwta-aux]. This functionality should belong in GroupPathService. 

As for the return of the update function, it should be just GroupingTimestampResult rather than a list. I suggest updating the GroupingTimestampResult to hold a list of groupingPaths.

Reply:
I made GroupingTimestampResults and I also needed to make UpdatedTimestampResults. Pretty much everything inside had to be turned into a list. The inherited method getGroupPath() doesn't do anything because it holds a list now instead of just a string.